projects/storefrontlib/src/cms-components/asm/services/asm-enabler.service.ts

```diff
@@ -79,7 +83,7 @@ export class AsmEnablerService {
     const factory = this.componentFactoryResolver.resolveComponentFactory(
       AsmMainUiComponent
     );
-    this.outletService.add('cx-storefront', factory, OutletPosition.BEFORE);
+    this.triggerService.renderDialog(this.triggerConfig.trigger?.asm, factory);
```

Comment:
What about handing over the triggerType, that can be used by the triggerService to lookup the configuration for it? I like to end up with a solution where only one depend service is required by the caller.
A bit similar to what we do with IconTypes